File: src/main/java/com/example/liftoff/storage/web/OneDriveStorage.java
package com.example.liftoff.storage.web;

import com.example.liftoff.storage.Storage;

import java.io.InputStream;
import java.nio.file.Path;
import java.util.Optional;

public class OneDriveStorage implements Storage {
    private final Path rootDirectory;
    private final String accessToken;

    public OneDriveStorage(Path rootDirectory, String accessToken) {
        this.rootDirectory = rootDirectory;
        this.accessToken = accessToken;
    }

    @Override
    public void storeFile(InputStream file, String fileName, Optional<String> fileDirectoryName) throws Exception {
        // TODO implement OneDrive storage functionality
    }
}

File: src/main/scala/com/example/liftoff/storage/web/OneDriveStorage.scala
package com.example.liftoff.storage.web

import com.example.liftoff.error.StorageError
import com.example.liftoff.storage.Storage
import zio.{IO, ULayer, ZLayer}

import java.io.InputStream

class OneDriveStorage extends Storage {

  override def storeFile(file: InputStream, fileName: String, fileDirectoryName: Option[String]): IO[StorageError, Unit] = ???
}

object OneDriveStorage {
  val layer: ULayer[Storage] = ZLayer.succeed(new OneDriveStorage)
}
